### PR TITLE
feat: add haproxy docker role

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR adds a role to install haproxy on docker as a replacement for Caddy.

### CODE REVIEW

File: roles/docker_haproxy/templates/haproxy.cfg.j2 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+backend https_k8s_servers
+  balance roundrobin
+  {% for backend in DOCKER_HAPROXY_BACKENDS %}
+  server s1 {{backend}}:{{DOCKER_HAPROXY_HTTPS_PORT}} check send-proxy
+  {% endfor %}
+  mode tcp
+
+frontend http
+  bind *:80 accept-proxy
```
Comment (MoisesGSalas) [inline roles/docker_haproxy/templates/haproxy.cfg.j2:38]: We probably won't need this because we are receiving traffic directly from the client (browser).

Comment (Ian2012) [reply roles/docker_haproxy/templates/haproxy.cfg.j2:38]: It's needed for the initial cert bot challenge request

Comment (MoisesGSalas) [reply roles/docker_haproxy/templates/haproxy.cfg.j2:38]: The `accept-proxy` argument? it fails if we only bind 80 without that?

Comment (Ian2012) [reply roles/docker_haproxy/templates/haproxy.cfg.j2:38]: I was confused, I tought you were talking about the :80 bind

File: roles/docker_haproxy/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    name: haproxy
+    image: "{{ DOCKER_HAPROXY_IMAGE }}"
+    restart_policy: unless-stopped
+    networks:
+      - name: "{{ DOCKER_HAPROXY_NETWORK }}"
+    state: started
+    published_ports:
+      - "{{ DOCKER_HAPROXY_BIND_ADDRESS }}:80:80"
+      - "{{ DOCKER_HAPROXY_BIND_ADDRESS }}:443:443"
+      - "{{ DOCKER_HAPROXY_BIND_ADDRESS }}:8404:8404"
```
Comment (MoisesGSalas) [inline roles/docker_haproxy/tasks/main.yml:33]: I think we should not expose this one to the whole internet. We can probably use a bind_address for the local network (like `10.0.1.0`). Check that if you do that the port is only accesible from the internal network.

**Review by MoisesGSalas — APPROVED** [review]

**Pull request merged**